### ISSUE TITLE
lint: typeUnparen false positive for method expressions

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#312]
This code triggers a warning:
```go
method := (*Client).set
```
It shouldn't.
